src/sentry/models/eventattachment.py

```diff
@@ -131,6 +131,13 @@ def getfile(self) -> IO[bytes]:
         elif self.blob_path.startswith("eventattachments/v1/"):
             storage = get_storage()
             compressed_blob = storage.open(self.blob_path)
+
+            # We want to log the compressed size here but we want to stream the payload.
+            # So, we swallow the cost of a metadata fetch.
+            compressed_blob.reload()
+            metric_emitter = StorageMetricEmitter("get", "attachments")
+            metric_emitter.record_compressed_size(compressed_blob.reload(), "gzip")
```

Comment:
you are calling `reload` twice here. I believe the second one should rather be a read of `.size`?

Also, since all these storage backends are pluggable, I’m not entirely sure this is even implemented for the filestore-go client? you would need to double-check this in the getsentry repo where the code lives.